src/XIVLauncher/Windows/SettingsControl.xaml.cs

```diff
@@ -389,6 +390,84 @@ private void GamePathEntry_OnTextChanged(object sender, TextChangedEventArgs e)
+            selectConfImportDialog.InitialDirectory = Paths.RoamingPath;
+            selectConfImportDialog.DefaultExt = ".xlconf";
+            selectConfImportDialog.Filter = "XLConfig Export|*.xlconf";
+
+            // Show open file dialog box
+            bool? result = selectConfImportDialog.ShowDialog();
+
+            // Process open file dialog box results
+            if (result == true)
+            {
+                try
+                {
+                    // Open document
+                    string zipFile = selectConfImportDialog.FileName;
+
+                    //TODO: Localize this text like a good programmer
+                    
+                    var importConfirmPromptResult = CustomMessageBox.Show(Loc.Localize("ConfigImportPrompt1", "XIVLauncher is going to import config from the file you selected.\n\n") + zipFile + Loc.Localize("ConfigImportPrompt2", "\n\nThis will overwrite any files that already exist.\n\nDo you want to proceed?"),
+                        Loc.Localize("ConfigImportPromptTitle", "Import confg file?"), MessageBoxButton.YesNo);
+
+                    if (importConfirmPromptResult == MessageBoxResult.Yes)
+                    {
+                        var window = new ConfigImportExportProgressWindow();
+                        window.SetTitle(Loc.Localize("ConfigImportRunning", "Running config import..."));
+
+                        Task.Run(() => DalamudAndPluginConfigImportExport.ImportConfig(zipFile, Paths.RoamingPath)).ContinueWith(async task =>
+                        {
+                            await task;
+                            window.Dispatcher.Invoke(() => window.Close());
+
+                            CustomMessageBox.Show(Loc.Localize("ConfigImportSuccess", "Dalamud and Plugin configuration import completed."), "XIVLauncher", MessageBoxButton.OK, MessageBoxImage.Information);
+                        });
+
+                        window.ShowDialog();
+                    }
+                    if (importConfirmPromptResult == MessageBoxResult.No)
+                    {
+                        CustomMessageBox.Show(Loc.Localize("ConfigImportCancelled", "Dalamud and Plugin configuration import cancelled."), "XIVLauncher", MessageBoxButton.OK, MessageBoxImage.Information);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    CustomMessageBox.Show(ex.ToString(), "XIVLauncher Error", MessageBoxButton.OK, MessageBoxImage.Error);
+                }
+
+            }
+        }
+        private void ExportConfigButton_OnClick(object sender, RoutedEventArgs e)
+        {
+            try
+            {
+                var window = new ConfigImportExportProgressWindow();
+                window.SetTitle(Loc.Localize("ConfigExportRunning", "Running config export..."));
+
+                Task.Run(async () => await DalamudAndPluginConfigImportExport.ExportConfig(Paths.RoamingPath)).ContinueWith(async task =>
```

Comment:
This task needs a continuation. You need to observe the exception if it didn't complete successfully and hide the progress window if it's still open.